k8_cortx_cloud/get-logs-from-pvc.sh

```diff
@@ -0,0 +1,112 @@
+#!/bin/bash
+
+function usage()
+{
+  echo -e "\n** Recover contents of PVC from Non-Running CORTX Containers **\n"
+  echo -e "Usage: \`sh $0 [PVC]\`\n"
+  echo "Optional Arguments:"
+  echo "    -n|--namespace NAMESPACE: K8s namespace that PVC is in (default=default)"
+  echo "    -f|--force:  Force overwrite of output file"
+  exit 1
+}
+
+pvc=
+namespace=default
+force_overwrite=false
+
+while [[ $# -gt 0 ]]; do
+  case $1 in
+    -n|--namespace )
+      namespace="$2"
+      shift 2
+      ;;
+    -f|--force )
+      force_overwrite=true
+      shift 1
+      ;;
+    * )
+      if [[ $1 = -* ]]; then
+        echo "ERROR: Unsupported Option \"$1\"."
+        usage
+      elif [[ -z "${pvc}" ]]; then
+        pvc=$1
+      fi
+      shift 1
+      ;;
+  esac
+done
+
+if [[ -z "${pvc}" ]]; then
+  echo "ERROR: Specify PVC."
+  exit 1
+fi
+
+datestr=$(date '+%Y%m%d.%H%M%S')
+job_name="cortx-log-${pvc}-${datestr}"
+tarfile="${pvc}.tgz"
+
+if [[ -f "${tarfile}" && "${force_overwrite}" == "false" ]]; then
+  printf "%s already exists. -f to overwrite.\n" "${tarfile}"
+  exit 1
+fi
+
+
+cat << EOF | kubectl apply -f - || true
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: "${job_name}"
+  namespace: "${namespace}"
+spec:
+  template:
+    spec:
+      containers:
+        - image: busybox
+          name: tar-pvc
+          command:
+            - sh
+            - -c
+            - tar cfz /tmp/"${tarfile}" -C /etc "${pvc}"; \
+              touch /tmp/tarfile_created; \
+              while [ ! -f /tmp/stopme ]; do sleep 1; done
+          volumeMounts:
+            - mountPath: "/etc/${pvc}"
+              name: data
+      volumes:
+        - name: data
+          persistentVolumeClaim:
+            claimName: "${pvc}"
+      restartPolicy: OnFailure
+EOF
+
+
+# Get pod name
+pod_status="Starting"
+while [[ "${pod_status}" != "Running" ]]; do
```

Comment:
You'd still need to get the pod name, but I think you could just skip the manual loop and checking status otherwise, and it would also put a time limit on the wait, which the manual loop doesn't have.

Reply:
I tested `--pod-running-timeout`, and it doesn't work very well.

I'd suggest `kubectl wait --for=condition=ready --selector=job-name="${job_name}" --namespace "${namespace}" pod` instead to replace this while loop.